numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,141 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
```

Comment:
```suggestion
@set_module('numpy')
def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
```

This should fix the CI failure.
Note that `set_module` will also have to be imported from `numpy.core.overrides`.

Reply:
Or perhaps we should add the full array_function dispatching

Reply:
I'm sorry, I don't know what that means. Could you please clarify? In the meantime, I'll get to work on those error tests...

Reply:
Ok, I read (parts of) NEP-18 and took a stab at adding the array_function dispatching. One thing that I am still unsure about is the usage of defaults for arguments. Please have a look and let me know what you think.